Refactor user auth to use SQLAlchemy ORM

maximvl opened the pull request:
Идея в том чтобы перенести наш код в /src/domain/ и разделить на models, api, queries, serializers чтобы более читабельно и легче было с ним работать. Работать с базой через sqlalchemy и pydantic для сериализации и валидации.

Replace direct DB client user functions with SQLAlchemy models and queries located in the new `src/users` module. Update the `/login` endpoint accordingly.

Centralize Flask extensions (SQLAlchemy, Bcrypt) in `src/core`. Improve error logging details in background jobs and app hooks. Comment out `/api/player_move` endpoint temporarily.

maximvl committed:
Refactor user auth to use SQLAlchemy ORM

Replace direct DB client user functions with SQLAlchemy models and
queries located in the new `src/users` module. Update the `/login`
endpoint accordingly.

Centralize Flask extensions (SQLAlchemy, Bcrypt) in `src/core`.
Improve error logging details in background jobs and app hooks.
Comment out `/api/player_move` endpoint temporarily.

maximvl commented on background_jobs.py:
```diff
@@ -175,13 +175,10 @@ def refresh_stream_statuses():
-                        + str(e)
-                        + ", line: "
-                        + str(exc_tb.tb_lineno)
+                        f"Stream check failed for {player_name},: {e}, line: {line}"
```
пара фиксов ворнингов

maximvl commented on db_client/db_client.py:
```diff
@@ -49,145 +49,6 @@ def conn(self):
             self.connection = MySQLdb.connect(**MYSQLCONF)
             return self.connection
 
-    # --- Методы для работы с таблицей users ---
```
перенес в /src/users/queries

maximvl commented on only_api_app.py:
```diff
@@ -19,27 +18,27 @@
     app.config["DEBUG"] = True
     app.config["JSON_AS_ASCII"] = False
-    app.json.ensure_ascii = False
+    # app.json.ensure_ascii = False
```
тоже жалуется что такого свойства нету, настройки на строчку выше должно хватать

olegsvs replied:
Кто жалуется? Без неё с русским текстом траблы в ответах. У тебя версия flask какая на тест стенде? https://github.com/aukus-devs/aukus-backend/blob/v2025-prod/requirements.txt#L5 ?

maximvl replied:
фласк и другие пакеты из requirements.txt

```
Pyright: Cannot assign to attribute "ensure_ascii" for class "JSONProvider"
  Attribute "ensure_ascii" is unknown
```

а откуда инфа что эта настройка работает?

maximvl commented on only_api_app.py:
```diff
@@ -19,27 +18,27 @@
 MYSQL_PASSWORD = os.getenv("MYSQL_PASSWORD")
 MYSQL_HOST = os.getenv("MYSQL_HOST")
 app = Flask(__name__)
-bcrypt = Bcrypt(app)
```
переместил в отдельный файл чтобы импортить оттуда и сюда и в апи

maximvl commented on only_api_app.py:
```diff
@@ -19,27 +18,27 @@
-    app.register_blueprint(auth_bp)
+    bcrypt.init_app(app)
+
+    app.register_blueprint(users_bp)
```
заменил auth_bp на новый из /src/users

maximvl commented on routes/api/login/login.py:
```diff
@@ -1,38 +0,0 @@
-from flask import (
```
переместил в /src/users/api

maximvl commented on routes/api/player/player.py:
```diff
@@ -134,88 +134,88 @@ def get_players():
     return jsonify({"players": players})
 
 
-@player_bp.route("/api/player_move", methods=["POST"])
```
пока закомментил потому что использует запросы к users, потом перемещу в /src/players/queries

maximvl committed:
Move more queries for users and moves

olegsvs committed:
Fix flask session reusing db

maximvl committed:
Feat: Add Games and Players API endpoints

Add /api/games endpoint to search games using data from an IGDB
database.

Add /api/players endpoint to retrieve detailed player information,
including map position, current game details with cover image, and
stream status.

Remove unused and commented-out methods from DatabaseClient.

maximvl committed:
Add pydantic dependency
